modules/ldapbackend/dnssec.cc

```diff
@@ -0,0 +1,1387 @@
+{
+  try {
+    std::string filter = "(&(objectClass=PdnsTSIGKey)(cn=" + d_pldap->escape( name.toString( "" ) ) + "))";
+    const char* attributes[] = { (char*)"PdnsKeyAlgorithm", (char*)"PdnsKeyContent", NULL };
+    PowerLDAP::sentry_t result;
+    PowerLDAP::SearchResult::Ptr search = d_pldap->search( "ou=TSIGKeys," + getArg( "metadata-searchdn" ), LDAP_SCOPE_ONELEVEL, filter, attributes );
+
+    if ( !search->getNext( result ) )
+      return false;
+
+    if ( algorithm->empty() || *algorithm == DNSName( result["PdnsKeyAlgorithm"][0] ) ) {
+      *algorithm = DNSName( result["PdnsKeyAlgorithm"][0] );
+      *content = result["PdnsKeyContent"][0];
+    }
+
+    return !content->empty();
+  }
+  catch( LDAPTimeout &lt )
+  {
+    g_log << Logger::Warning << d_myname << " Unable to search LDAP directory: " << lt.what() << endl;
+    throw DBException( "LDAP server timeout" );
+  }
+  catch( LDAPNoConnection &lnc )
+  {
+    g_log << Logger::Warning << d_myname << " Connection to LDAP lost, trying to reconnect" << endl;
+    if ( reconnect() )
+      return this->getTSIGKey( name, algorithm, content );
+    else
+      throw PDNSException( "Failed to reconnect to LDAP server" );
+  }
+  catch( LDAPException &le )
+  {
+    g_log << Logger::Error << d_myname << " Unable to search LDAP directory: " << le.what() << endl;
+    throw PDNSException( "LDAP server unreachable" );   // try to reconnect to another server
+  }
+  catch( std::exception &e )
+  {
+    g_log << Logger::Error << d_myname << " Caught STL exception retrieving TSIG key '" << name << "': " << e.what() << endl;
+    throw DBException( "STL exception" );
+  }
+}
+
+
+bool LdapBackend::setTSIGKey( const DNSName& name, const DNSName& algorithm, const string& content )
+{
+  try {
+    DNSName existingAlgo;
+    std::string existingContent;
+    if ( this->getTSIGKey( name, &existingAlgo, &existingContent ) ) {
+      LDAPMod algoMod;
+      algoMod.mod_op = LDAP_MOD_REPLACE;
+      algoMod.mod_type = (char*)"PdnsKeyAlgorithm";
+      std::string algoStr = algorithm.toString( "" );
+      char* algoValues[] = { (char*)(algoStr.c_str()), NULL };
+      algoMod.mod_values = algoValues;
+
+      LDAPMod contentMod;
+      contentMod.mod_op = LDAP_MOD_REPLACE;
+      contentMod.mod_type = (char*)"PdnsKeyContent";
+      char* contentValues[] = { (char*)(content.c_str()), NULL };
+      contentMod.mod_values = contentValues;
+
+      std::string filter = "(&(objectClass=PdnsTSIGKey)(cn=" + d_pldap->escape( name.toString( "" ) ) + "))";
+      const char* attributes[] = { (char*)"objectClass", NULL };
+      PowerLDAP::sentry_t result;
+      PowerLDAP::SearchResult::Ptr search = d_pldap->search( "ou=TSIGKeys," + getArg( "metadata-searchdn" ), LDAP_SCOPE_ONELEVEL, filter, attributes );
+
+      LDAPMod* mods[3];
+      mods[0] = &algoMod;
+      mods[1] = &contentMod;
+      mods[2] = NULL;
+
+      if ( !search->getNext( result, true ) ) // Whaaaa?
```

Comment:
Hi, for the time being (and hopefully going forward), pdns master is running the [check-spelling](https://github.com/marketplace/actions/check-spelling) action. In order to avoid a ❌ when merged https://github.com/jsoref/pdns/commit/2f8e9ae5b530d7687232deeb9e81d57da61d96ac#commitcomment-39230782 -- you'll need to merge/rebase on `master` to get the check (and the files) -- you can accept the advice (which should add various items to `expect.txt`)

Note: other there's one exception: I don't think `Whaaaa` belongs in the corpus ;-)